runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,391 @@
+                        clientDestinations = DestinationInfoVal.newBuilder().addAllDestinationIds(finalRemoteDestinations).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+                    return null;
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to add destinations, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Unable to add destinations", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove a destination from the LogReplicationModelMetadataTable for a specific group.
+     *
+     * @param logicalGroup The group of tables that are replicated, represented as a string.
+     * @param destination The cluster from which the logicalGroup should be removed from all future replication.
+     */
+    public void removeDestination(String logicalGroup, String destination) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(destination), String.format("[%s] %s", clientName, "destination is null or empty."));
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                    if (clientDestinations != null) {
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinations.getDestinationIdsList());
+
+                        if (clientDestinationIdsList.remove(destination)) {
+                            if (clientDestinationIdsList.size() == 0) {
+                                txn.delete(sourceMetadataTable, clientInfoKey);
+                            } else {
+                                clientDestinations = clientDestinations.toBuilder()
+                                        .clearDestinationIds()
+                                        .addAllDestinationIds(clientDestinationIdsList)
+                                        .build();
+                                txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                            }
+
+                            txn.commit();
+                            return null;
+                        }
+
+                        log.debug("[{}] {} {}", clientName, "No matching destination found for:", destination);
```

Comment:
nit: you can simplify and represent this as -
log.debug("[{}] No matching...for: {}", clientName, destination).  

Same for all other places